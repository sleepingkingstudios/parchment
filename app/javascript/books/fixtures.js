import { load } from '../fixtures';

/* eslint-disable-next-line import/prefer-default-export */
export const booksData = load('books');
