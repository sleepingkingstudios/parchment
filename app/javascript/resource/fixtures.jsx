import React from 'react';

export const Block = () => (<div />);
export const Form = () => (<div />);
export const Table = () => (<div />);

export const CreatePage = () => (<div />);
export const IndexPage = () => (<div />);
export const LandingPage = () => (<div />);
export const ShowPage = () => (<div />);
export const UpdatePage = () => (<div />);

export const fixtures = {
  create: {
    component: CreatePage,
    exact: true,
    options: { baseUrl: '/widgets', resourceName: 'widgets' },
    path: '/create',
    reducer: ((state = { widget: {} }) => state),
  },
  index: {
    component: IndexPage,
    exact: true,
    options: { baseUrl: '/widgets', resourceName: 'widgets' },
    path: '/',
    reducer: ((state = { widgets: [] }) => state),
  },
  show: {
    component: ShowPage,
    exact: true,
    options: { baseUrl: '/widgets', resourceName: 'widgets' },
    path: '/:id',
    reducer: ((state = { widget: {} }) => state),
  },
  update: {
    component: UpdatePage,
    exact: true,
    options: { baseUrl: '/widgets', resourceName: 'widgets' },
    path: '/:id/update',
    reducer: ((state = { widget: {} }) => state),
  },
};
