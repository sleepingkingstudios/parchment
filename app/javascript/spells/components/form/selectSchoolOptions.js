const selectSchoolsOptions = [
  {
    label: 'Abjuration',
    value: 'abjuration',
  },
  {
    label: 'Conjuration',
    value: 'conjuration',
  },
  {
    label: 'Divination',
    value: 'divination',
  },
  {
    label: 'Enchantment',
    value: 'enchantment',
  },
  {
    label: 'Evocation',
    value: 'evocation',
  },
  {
    label: 'Illusion',
    value: 'illusion',
  },
  {
    label: 'Necromancy',
    value: 'necromancy',
  },
  {
    label: 'Transmutation',
    value: 'transmutation',
  },
];

export default selectSchoolsOptions;
