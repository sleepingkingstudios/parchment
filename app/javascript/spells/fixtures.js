/* eslint-disable-next-line import/prefer-default-export */
export const spellsData = [
  {
    id: 'e17bca01-24d3-48d3-a308-099dd6dca373',
    level: 0,
    name: 'Embers',
    school: 'Evocation',
  },
  {
    id: 'e3641970-fc98-4674-b873-5532d89d9f26',
    level: 3,
    name: "Dragon's Breath",
    school: 'Evocation',
  },
  {
    id: 'a74d8c21-607d-4a3f-8cfd-2bbb48ec196a',
    level: 9,
    name: 'Wrath of the Titans',
    school: 'Evocation',
  },
];
