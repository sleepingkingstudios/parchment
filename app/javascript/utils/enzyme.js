import { valueOrDefault } from './object';
import { safeCapitalize } from './string';

export const extractInjectedComponent = (component) => {
  const injected = component.shallow().find('ComponentWithInjectedProps');

  return injected.exists() ? injected : component;
};

export const toHaveFormControl = expect => (received, expectedControl, withProps) => {
  const formControl = received.find(expectedControl);
  const expectedProps = valueOrDefault(withProps, {});

  expect(formControl).toExist();
  expect(formControl).toHaveProp(expectedProps);

  return { pass: true, message: () => '' };
};

export const toHaveFormGroup = expect => (received, expectedGroup, fn) => {
  const formGroup = received.find(expectedGroup);

  expect(formGroup).toExist();

  fn(extractInjectedComponent(formGroup).shallow());

  return { pass: true, message: () => '' };
};

export const toHaveFormInput = expect => (received, props) => {
  const {
    form,
    inputType,
    prop,
  } = props;

  expect(received).toHaveProp({ prop });

  const formInput = received.find(`${safeCapitalize(prop)}Input`);

  expect(formInput).toExist();
  expect(formInput).toHaveProp({ form });

  const shallowInput = formInput.shallow();

  expect(shallowInput).toExist();
  expect(shallowInput).toHaveDisplayName(valueOrDefault(inputType, 'FormInput'));

  return { pass: true, message: () => '' };
};
