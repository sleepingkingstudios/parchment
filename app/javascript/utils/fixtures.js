/* eslint-disable-next-line import/prefer-default-export */
export const trimMultilineLiteral = (text) => {
  if (!text) { return ''; }

  const paragraphs = text.split('\n\n');

  return paragraphs
    .map((paragraph) => {
      const lines = paragraph.split('\n');

      return lines.map(line => line.trim()).join(' ');
    })
    .join('\n\n');
};
