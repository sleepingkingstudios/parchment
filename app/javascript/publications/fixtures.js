import { load } from '../fixtures';

const isOfficial = (publication) => {
  const { playtest, publisherName } = publication;

  return !playtest && publisherName === 'Wizards of the Coast';
};

const rawData = load('publications');

/* eslint-disable-next-line import/prefer-default-export */
export const publicationsData = rawData
  .map(publication => Object.assign(
    {},
    publication,
    {
      official: isOfficial(publication),
    },
  ));
