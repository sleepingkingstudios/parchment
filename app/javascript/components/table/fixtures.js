import { humanizeList } from '../../utils/array';

export const formatCrewCapacity = (item) => {
  const count = item.crewCapacity;

  if (count === 1) { return '1 Kerbal'; }

  return `${count} Kerbals`;
};

export const formatExperiments = (item) => {
  humanizeList(item.experiments, { default: 'None' });
};

export const columns = [
  {
    label: 'Name',
    prop: 'name',
  },
  {
    label: 'Crew Capacity',
    prop: 'crewCapacity',
    value: formatCrewCapacity,
  },
  {
    label: 'Experiments',
    prop: 'experiments',
  },
];

export const rocketsData = [
  {
    id: '83f5e9e8-92df-4203-bc2e-4381370c2186',
    name: 'Apprentice',
    crewCapacity: 1,
    experiments: [
      'Crew Report',
      'Mystery Goo',
      'Temperature Scan',
      'Pressure Scan',
    ],
  },
  {
    id: '4bd7454c-3b15-4047-8d57-dfa831dac58e',
    name: 'Sorcerer',
    crewCapacity: 2,
    experiments: [
      'Crew Report',
      'Materials Study',
      'Mystery Goo',
      'Temperature Scan',
      'Pressure Scan',
    ],
  },
  {
    id: '2fead083-d9ea-4f04-949e-d8c00e800cbb',
    name: 'Warlock',
    crewCapacity: 3,
    experiments: [
      'Crew Report',
      'Materials Study',
      'Mystery Goo',
      'Temperature Scan',
      'Pressure Scan',
      'Gravity Scan',
    ],
  },
];
