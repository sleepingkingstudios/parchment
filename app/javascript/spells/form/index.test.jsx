import React from 'react';
import { shallow, mount } from 'enzyme';

import SpellForm, {
  LevelField,
  NameField,
  RitualField,
  SchoolField,
  CastingTimeField,
  RangeField,
  DurationField,
  DescriptionField,
  SubmitButton,
} from './index';

import { spellsData } from '../fixtures';
import selectSchoolOptions from './selectSchoolOptions';

describe('<SpellForm />', () => {
  const onChangeAction = jest.fn(({ propName, value }) => ({ payload: { propName, value } }));
  const onSubmitAction = jest.fn(() => ({ ok: true }));
  const spell = spellsData[0];
  const defaultProps = {
    onChangeAction,
    onSubmitAction,
    spell,
  };
  const form = {
    data: spell,
    namespace: 'spell',
    onChangeAction,
    onSubmitAction,
  };

  it('should render a form', () => {
    const rendered = shallow(<SpellForm {...defaultProps} />);

    expect(rendered).toHaveDisplayName('Form');
    expect(rendered).toHaveProp({ form });
  });

  it('should render the name input field', () => {
    expect(NameField.inputDisplayName).toEqual('FormInput');

    const rendered = shallow(<SpellForm {...defaultProps} />);
    const input = rendered.find('NameField');

    expect(input).toExist();
    expect(input).toHaveProp({ form });
  });

  it('should render the school select field', () => {
    expect(SchoolField.inputDisplayName).toEqual('FormSelectInput');

    const rendered = shallow(<SpellForm {...defaultProps} />);
    const input = rendered.find('SchoolField');
    const options = selectSchoolOptions;

    expect(input).toExist();
    expect(input).toHaveProp({ form });
    expect(input).toHaveProp({ options });
  });

  it('should render the level input field', () => {
    expect(LevelField.inputDisplayName).toEqual('FormNumericInput');

    const rendered = shallow(<SpellForm {...defaultProps} />);
    const input = rendered.find('LevelField');

    expect(input).toExist();
    expect(input).toHaveProp({ form });
  });

  it('should render the ritual checkbox field', () => {
    expect(RitualField.inputDisplayName).toEqual('FormCheckboxInput');

    const rendered = shallow(<SpellForm {...defaultProps} />);
    const input = rendered.find('RitualField');

    expect(input).toExist();
    expect(input).toHaveProp({ form });
  });

  it('should render the casting time input field', () => {
    expect(CastingTimeField.inputDisplayName).toEqual('FormInput');

    const rendered = shallow(<SpellForm {...defaultProps} />);
    const input = rendered.find('CastingTimeField');

    expect(input).toExist();
    expect(input).toHaveProp({ form });
  });

  it('should render the range input field', () => {
    expect(RangeField.inputDisplayName).toEqual('FormInput');

    const rendered = shallow(<SpellForm {...defaultProps} />);
    const input = rendered.find('RangeField');

    expect(input).toExist();
    expect(input).toHaveProp({ form });
  });

  it('should render the duration input field', () => {
    expect(DurationField.inputDisplayName).toEqual('FormInput');

    const rendered = shallow(<SpellForm {...defaultProps} />);
    const input = rendered.find('DurationField');

    expect(input).toExist();
    expect(input).toHaveProp({ form });
  });

  it('should render the components field', () => {
    const rendered = shallow(<SpellForm {...defaultProps} />);
    const input = rendered.find('SpellFormComponentsField');

    expect(input).toExist();
    expect(input).toHaveProp({ form });
  });

  it('should render the description textarea field', () => {
    expect(DescriptionField.inputDisplayName).toEqual('FormTextAreaInput');

    const rendered = shallow(<SpellForm {...defaultProps} />);
    const input = rendered.find('DescriptionField');

    expect(input).toExist();
    expect(input).toHaveProp({ form });
  });

  it('should render the submit button', () => {
    expect(SubmitButton.inputDisplayName).toEqual('Button');

    const rendered = shallow(<SpellForm {...defaultProps} />);
    const input = rendered.find('SubmitButton');

    expect(input).toExist();
    expect(input).toHaveProp({ form });
    expect(input).toHaveProp('children', 'Create Spell');
  });

  it('should match the snapshot', () => {
    const rendered = mount(<SpellForm {...defaultProps} />);

    expect(rendered).toMatchSnapshot();
  });

  describe('with isUpdate: true', () => {
    const props = { ...defaultProps, isUpdate: true };

    it('should render the submit button', () => {
      expect(SubmitButton.inputDisplayName).toEqual('Button');

      const rendered = shallow(<SpellForm {...props} />);
      const input = rendered.find('SubmitButton');

      expect(input).toExist();
      expect(input).toHaveProp({ form });
      expect(input).toHaveProp('children', 'Update Spell');
    });

    it('should match the snapshot', () => {
      const rendered = mount(<SpellForm {...props} />);

      expect(rendered).toMatchSnapshot();
    });
  });
});
