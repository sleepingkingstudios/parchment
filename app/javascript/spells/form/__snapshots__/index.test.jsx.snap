// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SpellForm /> should match the snapshot 1`] = `
<Form
  className="spell-form"
  form={
    Object {
      "data": Object {
        "spell": Object {
          "castingTime": "1 reaction, which you take when you speak your own name or hear it spoken",
          "description": "Immediately after speaking or hearing your name, you create the illusory sound of an ominous, distant thunderclap.",
          "duration": "Instantaneous",
          "id": "e17bca01-24d3-48d3-a308-099dd6dca373",
          "level": 0,
          "materialComponent": "",
          "name": "Thunderclap",
          "range": "Self",
          "ritual": false,
          "school": "illusion",
          "shortDescription": "Create an ominous thunderclap when your name is spoken",
          "somaticComponent": false,
          "verbalComponent": false,
        },
      },
      "errors": Object {},
      "onChangeAction": [MockFunction],
      "onSubmitAction": [MockFunction],
      "path": Array [
        "spell",
      ],
    }
  }
>
  <FormRow
    align="left"
    className=""
    verticalAlign="top"
  >
    <NameField
      colWidth="12"
      form={
        Object {
          "data": Object {
            "spell": Object {
              "castingTime": "1 reaction, which you take when you speak your own name or hear it spoken",
              "description": "Immediately after speaking or hearing your name, you create the illusory sound of an ominous, distant thunderclap.",
              "duration": "Instantaneous",
              "id": "e17bca01-24d3-48d3-a308-099dd6dca373",
              "level": 0,
              "materialComponent": "",
              "name": "Thunderclap",
              "range": "Self",
              "ritual": false,
              "school": "illusion",
              "shortDescription": "Create an ominous thunderclap when your name is spoken",
              "somaticComponent": false,
              "verbalComponent": false,
            },
          },
          "errors": Object {},
          "onChangeAction": [MockFunction],
          "onSubmitAction": [MockFunction],
          "path": Array [
            "spell",
          ],
        }
      }
    />
    <SchoolField
      colWidth="8"
      form={
        Object {
          "data": Object {
            "spell": Object {
              "castingTime": "1 reaction, which you take when you speak your own name or hear it spoken",
              "description": "Immediately after speaking or hearing your name, you create the illusory sound of an ominous, distant thunderclap.",
              "duration": "Instantaneous",
              "id": "e17bca01-24d3-48d3-a308-099dd6dca373",
              "level": 0,
              "materialComponent": "",
              "name": "Thunderclap",
              "range": "Self",
              "ritual": false,
              "school": "illusion",
              "shortDescription": "Create an ominous thunderclap when your name is spoken",
              "somaticComponent": false,
              "verbalComponent": false,
            },
          },
          "errors": Object {},
          "onChangeAction": [MockFunction],
          "onSubmitAction": [MockFunction],
          "path": Array [
            "spell",
          ],
        }
      }
      options={
        Array [
          Object {
            "label": "Abjuration",
            "value": "abjuration",
          },
          Object {
            "label": "Conjuration",
            "value": "conjuration",
          },
          Object {
            "label": "Divination",
            "value": "divination",
          },
          Object {
            "label": "Enchantment",
            "value": "enchantment",
          },
          Object {
            "label": "Evocation",
            "value": "evocation",
          },
          Object {
            "label": "Illusion",
            "value": "illusion",
          },
          Object {
            "label": "Necromancy",
            "value": "necromancy",
          },
          Object {
            "label": "Transmutation",
            "value": "transmutation",
          },
        ]
      }
    />
    <LevelField
      colWidth="2"
      form={
        Object {
          "data": Object {
            "spell": Object {
              "castingTime": "1 reaction, which you take when you speak your own name or hear it spoken",
              "description": "Immediately after speaking or hearing your name, you create the illusory sound of an ominous, distant thunderclap.",
              "duration": "Instantaneous",
              "id": "e17bca01-24d3-48d3-a308-099dd6dca373",
              "level": 0,
              "materialComponent": "",
              "name": "Thunderclap",
              "range": "Self",
              "ritual": false,
              "school": "illusion",
              "shortDescription": "Create an ominous thunderclap when your name is spoken",
              "somaticComponent": false,
              "verbalComponent": false,
            },
          },
          "errors": Object {},
          "onChangeAction": [MockFunction],
          "onSubmitAction": [MockFunction],
          "path": Array [
            "spell",
          ],
        }
      }
    />
    <RitualField
      colWidth="2"
      form={
        Object {
          "data": Object {
            "spell": Object {
              "castingTime": "1 reaction, which you take when you speak your own name or hear it spoken",
              "description": "Immediately after speaking or hearing your name, you create the illusory sound of an ominous, distant thunderclap.",
              "duration": "Instantaneous",
              "id": "e17bca01-24d3-48d3-a308-099dd6dca373",
              "level": 0,
              "materialComponent": "",
              "name": "Thunderclap",
              "range": "Self",
              "ritual": false,
              "school": "illusion",
              "shortDescription": "Create an ominous thunderclap when your name is spoken",
              "somaticComponent": false,
              "verbalComponent": false,
            },
          },
          "errors": Object {},
          "onChangeAction": [MockFunction],
          "onSubmitAction": [MockFunction],
          "path": Array [
            "spell",
          ],
        }
      }
    />
    <CastingTimeField
      colWidth="4"
      form={
        Object {
          "data": Object {
            "spell": Object {
              "castingTime": "1 reaction, which you take when you speak your own name or hear it spoken",
              "description": "Immediately after speaking or hearing your name, you create the illusory sound of an ominous, distant thunderclap.",
              "duration": "Instantaneous",
              "id": "e17bca01-24d3-48d3-a308-099dd6dca373",
              "level": 0,
              "materialComponent": "",
              "name": "Thunderclap",
              "range": "Self",
              "ritual": false,
              "school": "illusion",
              "shortDescription": "Create an ominous thunderclap when your name is spoken",
              "somaticComponent": false,
              "verbalComponent": false,
            },
          },
          "errors": Object {},
          "onChangeAction": [MockFunction],
          "onSubmitAction": [MockFunction],
          "path": Array [
            "spell",
          ],
        }
      }
    />
    <RangeField
      colWidth="4"
      form={
        Object {
          "data": Object {
            "spell": Object {
              "castingTime": "1 reaction, which you take when you speak your own name or hear it spoken",
              "description": "Immediately after speaking or hearing your name, you create the illusory sound of an ominous, distant thunderclap.",
              "duration": "Instantaneous",
              "id": "e17bca01-24d3-48d3-a308-099dd6dca373",
              "level": 0,
              "materialComponent": "",
              "name": "Thunderclap",
              "range": "Self",
              "ritual": false,
              "school": "illusion",
              "shortDescription": "Create an ominous thunderclap when your name is spoken",
              "somaticComponent": false,
              "verbalComponent": false,
            },
          },
          "errors": Object {},
          "onChangeAction": [MockFunction],
          "onSubmitAction": [MockFunction],
          "path": Array [
            "spell",
          ],
        }
      }
    />
    <DurationField
      colWidth="4"
      form={
        Object {
          "data": Object {
            "spell": Object {
              "castingTime": "1 reaction, which you take when you speak your own name or hear it spoken",
              "description": "Immediately after speaking or hearing your name, you create the illusory sound of an ominous, distant thunderclap.",
              "duration": "Instantaneous",
              "id": "e17bca01-24d3-48d3-a308-099dd6dca373",
              "level": 0,
              "materialComponent": "",
              "name": "Thunderclap",
              "range": "Self",
              "ritual": false,
              "school": "illusion",
              "shortDescription": "Create an ominous thunderclap when your name is spoken",
              "somaticComponent": false,
              "verbalComponent": false,
            },
          },
          "errors": Object {},
          "onChangeAction": [MockFunction],
          "onSubmitAction": [MockFunction],
          "path": Array [
            "spell",
          ],
        }
      }
    />
    <SpellFormComponentsField
      colWidth="12"
      form={
        Object {
          "data": Object {
            "spell": Object {
              "castingTime": "1 reaction, which you take when you speak your own name or hear it spoken",
              "description": "Immediately after speaking or hearing your name, you create the illusory sound of an ominous, distant thunderclap.",
              "duration": "Instantaneous",
              "id": "e17bca01-24d3-48d3-a308-099dd6dca373",
              "level": 0,
              "materialComponent": "",
              "name": "Thunderclap",
              "range": "Self",
              "ritual": false,
              "school": "illusion",
              "shortDescription": "Create an ominous thunderclap when your name is spoken",
              "somaticComponent": false,
              "verbalComponent": false,
            },
          },
          "errors": Object {},
          "onChangeAction": [MockFunction],
          "onSubmitAction": [MockFunction],
          "path": Array [
            "spell",
          ],
        }
      }
    />
    <DescriptionField
      colWidth="12"
      form={
        Object {
          "data": Object {
            "spell": Object {
              "castingTime": "1 reaction, which you take when you speak your own name or hear it spoken",
              "description": "Immediately after speaking or hearing your name, you create the illusory sound of an ominous, distant thunderclap.",
              "duration": "Instantaneous",
              "id": "e17bca01-24d3-48d3-a308-099dd6dca373",
              "level": 0,
              "materialComponent": "",
              "name": "Thunderclap",
              "range": "Self",
              "ritual": false,
              "school": "illusion",
              "shortDescription": "Create an ominous thunderclap when your name is spoken",
              "somaticComponent": false,
              "verbalComponent": false,
            },
          },
          "errors": Object {},
          "onChangeAction": [MockFunction],
          "onSubmitAction": [MockFunction],
          "path": Array [
            "spell",
          ],
        }
      }
      rows={5}
    />
  </FormRow>
  <FormRow
    align="right"
    className=""
    verticalAlign="top"
  >
    <CancelButton
      colWidth="3"
      form={
        Object {
          "data": Object {
            "spell": Object {
              "castingTime": "1 reaction, which you take when you speak your own name or hear it spoken",
              "description": "Immediately after speaking or hearing your name, you create the illusory sound of an ominous, distant thunderclap.",
              "duration": "Instantaneous",
              "id": "e17bca01-24d3-48d3-a308-099dd6dca373",
              "level": 0,
              "materialComponent": "",
              "name": "Thunderclap",
              "range": "Self",
              "ritual": false,
              "school": "illusion",
              "shortDescription": "Create an ominous thunderclap when your name is spoken",
              "somaticComponent": false,
              "verbalComponent": false,
            },
          },
          "errors": Object {},
          "onChangeAction": [MockFunction],
          "onSubmitAction": [MockFunction],
          "path": Array [
            "spell",
          ],
        }
      }
      isUpdate={false}
    />
    <SubmitButton
      colWidth="3"
      disabled={false}
      form={
        Object {
          "data": Object {
            "spell": Object {
              "castingTime": "1 reaction, which you take when you speak your own name or hear it spoken",
              "description": "Immediately after speaking or hearing your name, you create the illusory sound of an ominous, distant thunderclap.",
              "duration": "Instantaneous",
              "id": "e17bca01-24d3-48d3-a308-099dd6dca373",
              "level": 0,
              "materialComponent": "",
              "name": "Thunderclap",
              "range": "Self",
              "ritual": false,
              "school": "illusion",
              "shortDescription": "Create an ominous thunderclap when your name is spoken",
              "somaticComponent": false,
              "verbalComponent": false,
            },
          },
          "errors": Object {},
          "onChangeAction": [MockFunction],
          "onSubmitAction": [MockFunction],
          "path": Array [
            "spell",
          ],
        }
      }
      isUpdate={false}
      requestStatus="initialized"
    />
  </FormRow>
</Form>
`;
